src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala

```diff
@@ -52,15 +52,26 @@ class KinesisS3Emitter(
 ) extends IEmitter[EmitterInput]  {
 
   val s3Emitter = new S3Emitter(s3Config, provider, badSink, maxConnectionTime, tracker)
-  val dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+  val calendar = Calendar.getInstance()
 
   /**
    * Determines the filename in S3, which is the corresponding
    * Kinesis sequence range of records in the file.
    */
-  protected def getBaseFilename(firstSeq: String, lastSeq: String): String =
-    dateFormat.format(Calendar.getInstance().getTime()) +
-      "-" + firstSeq + "-" + lastSeq
+  protected def getBaseFilename(firstSeq: String, lastSeq: String): String = {
+    val date = calendar.getTime()
+    val dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+    val (yearFormat, monthFormat, dayFormat) = ( new SimpleDateFormat("yyyy"), new SimpleDateFormat("MM"), new SimpleDateFormat("dd"))
+
+    var prefix = ""
```

Comment:
`if` statements return a value in Scala so you can change this to 

```
val prefix = if (s3Config.partitioningFormat == "flat") {
      s"${dateFormat.format(date)}-"
    } else if (s3Config.partitioningFormat == "hive") {
      s"Year=${yearFormat.format(date)}/Month=${monthFormat.format(date)}/Day=${dayFormat.format(date)}/"
    }
```
if you want to be more functional and Scala-y (ie not mutating variables)

Reply:
Or with `match` since we discussed that:
```
val prefix = s3Config.partitioningFormat match {
     case "flat" => s"${dateFormat.format(date)}-"
     case "hive" => s"Year=${yearFormat.format(date)}/Month=${monthFormat.format(date)}/Day=${dayFormat.format(date)}/"
    }
```

Reply:
Both could maybe do with a fallback to return an error message if something else ended up in that string. Either an `else` or a `case <variable name> => <error message about variable name>`. Or default to `"flat"` here too.